File: 11. Container With Most Water/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        left = 0;
        right = len(height)-1;
        t_max = 0

        while(left < right):
            if height[left] < height[right]:
                t_max = max(t_max, height[left]*(right-left))
                left += 1
            else:
                t_max = max(t_max, height[right]*(right-left))
                right -= 1
        
        return t_max
            

File: 11. Container With Most Water/11.java
class Solution {
    public int maxArea(int[] height) {

        int l = 0, r = height.length-1;
        int tmax = 0;
        while(l < r) {
            if(height[l] < height[r]) {
                tmax = Math.max(tmax, height[l]*(r-l));
                l++;
            }
            else{
                tmax = Math.max(tmax, height[r]*(r-l));
                r--;
            }

        }
        return tmax;
    }
}
